### ISSUE TITLE
Console traceback not always appearing

### ISSUE THREAD

**Issue by gtritchie** [rstudio/rstudio#7928]
### System details

    RStudio Edition : Desktop or Server
    RStudio Version : 1.4.887
    OS Version      : any
    R Version       : any

### Steps to reproduce the problem
- In 1.4, turn on the virtual scroller (Global Options / Console / Limit console display to a subset of total content)
- Create an R script with below code and source it. Then execute `f2()` again in the console.

```
f1 <- function() {
  stop("Inspect me.")
}

f2 <- function() {
  f1()
}

f2()
```
### Describe the problem in detail
- Plain text results for the error, no traceback UI
- With the virtual scroller mode OFF, sometimes the traceback UI appears, sometimes it doesn't (e.g. try running `f2()` over and over via up-arrow + Enter); haven't figured out the pattern here seems very inconsistent (timing?)

![screenshot of console showing incorrect traceback UI](https://user-images.githubusercontent.com/10569626/94727860-2c284d00-0314-11eb-97fd-90bc0fe1132c.png)

### Describe the behavior you expected
Console should show the traceback UI for the initial source and for the separate `f2()` invocation as seen here in 1.3.1093:

![screenshot of console showing correct traceback UI instances](https://user-images.githubusercontent.com/10569626/94726909-c4bdcd80-0312-11eb-9ec0-08a6201eaa92.png)

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting a RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by kfeinauer** [comment]
Attempted to verify on 1.4.944 - this appears to still be broken.

I reprod using Gary's file above.

Steps to break:

1. Set `Limit console display` option to 50, then I cleared my Console history. 
2. Source the script
3. Note that the traceback shows up - it appears to work correctly
4. Immediately source the script again - the traceback no longer appears
5. Continued sourcings fail to display the traceback

Another issue is that the traceback never seems to occur when the virtual scroller is turned off.

**Comment by kevinushey** [comment]
Looks to me like this was indeed fixed by https://github.com/rstudio/rstudio/pull/8294.

**Comment by kevinushey** [comment]
Assigning @kfeinauer to re-verify; please feel free to re-assign if you won't be able to get to this one.

**Comment by kfeinauer** [comment]
Verified fixed 1.4.1007
